src/launchpad/analyzers/ios.py

```diff
@@ -139,6 +135,43 @@ def _extract_app_info(self, xcarchive: ZippedXCArchive) -> IOSAppInfo:
             sdk_version=plist.get("DTSDKName"),
         )
 
+    def _detect_file_type(self, file_path: Path) -> str:
```

Comment:
Some files like Mach-O binaries don't have an explicit file extension. In these cases I added a fallback to use the `file` command and determine the type. I have some future cases in mind that depend on this too.